File: Python3/Find Words That Can Be Formed by Characters.py
#1
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        sum, chars_counter = 0, collections.Counter(chars)
        for word in words:
            word_counter = collections.Counter(word)
            for c in word_counter:
                if word_counter[c] > chars_counter[c]:
                    break
            else:
                sum += len(word)
        return sum

#2 all == > [true,true,true] = true, [true,false,true] = false
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        sum, ct = 0, collections.Counter
        chars_counter = ct(chars)
        for word in words:
            word_counter = ct(word)
            if all(word_counter[c] <= chars_counter[c] for c in word_counter):
                sum += len(word)
        return sum

File: Golang/Find Words That Can Be Formed by Characters.go
func countCharacters(words []string, chars string) int {
    charSet := make([]int, 26)
    
    for i := 0; i < len(chars); i++ {
        charSet[int(chars[i] - 'a')]++
    }
    
    sum := 0
    
    for _, word := range words {
        currentCharSet := make([]int, 26)
        copy(currentCharSet, charSet)
        
        isValid := true
        
        for i := 0; i < len(word); i++ {
            currentCharSet[int(word[i] - 'a')]--
            
            if currentCharSet[int(word[i] - 'a')] < 0 {
                isValid = false
                break
            }
        }
        
        if isValid {
            sum += len(word)
        }
    }
    
    return sum
}